test/logster/test_redis_store.rb

```diff
@@ -712,9 +727,32 @@ def test_rate_limits_with_prefix
     end
   end
 
+  def test_suppression_patterns_are_cached
+    @store.allow_custom_ignore = true
+    rec = Logster::SuppressionPattern.new(/forest/, store: @store)
+    rec.save
+
+    @store.report(Logger::INFO, "test", "littleforest")
+    latest = @store.latest
+    assert_equal(0, latest.size)
+
+    rec.destroy
+    @store.report(Logger::INFO, "test", "anotherforest")
+    assert_equal(0, @store.latest.size)
+
+    Time.stub :now, Time.at(Time.now + 3) do
+      @store.report(Logger::INFO, "test", "myforest")
+      latest = @store.latest
+      assert_equal(1, latest.size)
+      assert_equal("myforest", latest.first.message)
+    end
+  end
+
   private
 
   def reset_redis
-    @store.redis.flushall
+    @store.redis.keys.each do |key|
```

Comment:
my point here is actually a bit more that this. I would like us to use proper logster apis for clearing stuff. We should never be reaching to redis from the tests here, its an implementation detail. 

```
@store.clear_suppressions! 
@store.clear_logs!
```
or

```
@store.clear_all
```

Reply:
I've removed direct redis calls in tests as much as I can. There are only 2 places left:

1. https://github.com/discourse/logster/blob/master/test/logster/test_redis_rate_limiter.rb which seems to be fundamentally based on redis and pulling out redis calls from it is going to be very tricky.

2. https://github.com/discourse/logster/blob/5bc5014f66b059becf92508f56214abb7ef2f859/test/logster/test_redis_store.rb#L160-L193 these 2 tests were added when we separated env from main message body. The purpose of these tests is to ensure backward compatibility which requires a couple hacks. When I wrote these tests I added a note that these can be removed in 6-12 months. I can remove them now if you want.

Reply:
OK go ahead and remove the backwards compat test, we can just live with the rate limiting test for now, it is tricky to abstract into an interface.